main.go

```diff
@@ -28,6 +28,8 @@ func setupRouter() *gin.Engine {
 
 	engine.GET("/reports", report.GetReports)
 
+	engine.POST("/reports/new", report.AddReport)
```

Comment:
```suggestion
	engine.POST("/reports", report.AddReport)
```

In REST, using POST on a route infers that you are creating a new entry of whatever is in the route